### ISSUE TITLE
loss function for student model training

### ISSUE THREAD

**Issue by Xiao0728** [castorini/tct_colbert#1]
Hi,

I'm looking at the loss function for training the student model of tct-colbert, the loss function described in the paper is used as:
![image](https://user-images.githubusercontent.com/43675140/144316053-845740d6-c6f0-4d9e-bd28-ecb0c904d26f.png)
i.e. the loss for the student model contains two components: the first is the hard label loss and the second is the KL divergence between the sampled probability distributions between the teacher and student;
However, while I was looking at this code implementation: https://github.com/castorini/tct_colbert/blob/d9ab5628eda47a840628f3a39628c52b269f82da/train/model_fn.py#L142, I have found the following issues: 
1. the loss function for training the student model doesn't have the hard label loss component;
2. there is no gamma parameter implemented.

If I missed them somewhere else, please correct me. 

Xiao Wang

**Comment by jacklin64** [comment]
Hi, thanks for your question. Yes, you are right. In our more recent version, we find it doesn't need hard label. Please refer to Eq.(4) in our paper (https://aclanthology.org/2021.repl4nlp-1.17.pdf)
